src/transport/grpc/mod.rs

```diff
@@ -622,731 +646,856 @@ impl<'a> TryFrom<&'a Interfaces> for NodeData {
+        let interface = Interface::from_str(crate::test::E2E_SERVER_DATASTREAM).unwrap();
+        let interface_name = interface.interface_name().to_owned();
+        let expected_object = Value::Object((MockServerObject {}).astarte_aggregate().unwrap());
         let proto_payload: astarte_message_hub_proto::astarte_message::Payload =
             expected_object.into();
-
         let astarte_message = AstarteMessage {
-            interface_name: "org.astarte-platform.rust.examples.object-datastream.DeviceDatastream"
-                .to_string(),
-            path: "/1".to_string(),
+            interface_name: interface_name.clone(),
+            path: PATH.to_string(),
             timestamp: None,
             payload: Some(proto_payload.clone()),
         };
 
-        // Send object from server
-        channels
-            .server_response_sender
-            .send(vec![Ok(astarte_message.into())])
-            .await
-            .unwrap();
-
-        let interfaces =
-            Interfaces::from_iter([
-                Interface::from_str(crate::test::OBJECT_DEVICE_DATASTREAM).unwrap()
-            ]);
-
-        let client_connection = async {
-            let client = client_future.await;
-
-            mock_astarte_grpc_client(client, &interfaces).await
-        };
-
-        let (_client, mut connection) = tokio::select! {
-            _ = server_future => panic!("The server closed before the client could complete sending the data"),
-            res = client_connection => {
-                println!("Client connected correctly: {}", res.is_ok());
-
-                res.expect("Expected correct connection in test")
-            },
-        };
+        mock_client_rx
+            .expect_attach::<tonic::Request<astarte_message_hub_proto::Node>>()
+            .times(1)
+            .in_sequence(&mut seq)
+            .returning(move |_i| {
+                Ok(tonic::Response::new(setup_stream(vec![Ok(Some(
+                    astarte_message_hub_proto::MessageHubEvent {
+                        event: Some(
+                            astarte_message_hub_proto::message_hub_event::Event::Message(
+                                astarte_message.clone(),
+                            ),
+                        ),
+                    },
+                ))])))
+            });
+
+        let (_client, mut connection) =
+            mock_grpc(mock_client_tx, mock_client_rx, &Interfaces::new(), store)
+                .await
+                .unwrap();
 
-        expect_messages!(connection.next_event().await;
-            Some(ReceivedEvent {
-                ref interface,
-                ref path,
-                payload: GrpcPayload {
-                    data,
+        //let event = ;
+        let Some(ReceivedEvent {
+            interface: received_interface,
+            path: received_path,
+            payload:
+                GrpcPayload {
+                    data: received_proto_payload,
```

Comment:
I'd use a more readable sintax, like

```suggestion
        let Some(event) = connection.next_event().await.unwrap() else {
            panic!("Event received did not match the pattern");
        };
        
        assert_eq!(event.interface, interface_name);
        assert_eq!(event.path, PATH);
        assert_eq!(event.payload.data, proto_payload);
```